src/dmd/root/rmem.d

```diff
@@ -256,4 +256,22 @@ extern (D) static char[] xarraydup(const(char)[] s) nothrow
     return null;
 }
 
+/**
+Makes a copy of the given array on newly allocated memory.
+
+Params:
+    s = array to copy
 
+Returns: A copy of the input array.
+*/
+extern (D) static auto arraydup(T)(T[] s) nothrow
```

Comment:
`const scope` maybe?

Reply:
* `static` is not necessary
* please don’t use `auto` for the return type
* please use standard D naming conventions

Reply:
FYI: I copied the declaration of the previous `xarraydup`, hence the name.

> can this be made pure?

Yes, if the `pureMalloc` declarations are added. However, I don't see a big to do this in this PR as `xarraydup` isn't pure either, DMD can't take much advantage of purity anyhow and this PR still has other, real issues.